cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/cube/CubeDatasetDefinition.java

```diff
@@ -0,0 +1,195 @@
+
+    int[] resolutions = getResolutions(spec.getProperties());
+    for (int resolution : resolutions) {
+      admins.add(tableDef.getAdmin(datasetContext, spec.getSpecification(String.valueOf(resolution)), classLoader));
+    }
+
+    return new CompositeDatasetAdmin(admins);
+  }
+
+  @Override
+  public CubeDataset getDataset(DatasetContext datasetContext, DatasetSpecification spec,
+                                Map<String, String> arguments, ClassLoader classLoader) throws IOException {
+
+    Table entityTable = tableDef.getDataset(datasetContext, spec.getSpecification("entity"), arguments, classLoader);
+
+    int[] resolutions = getResolutions(spec.getProperties());
+    Map<Integer, Table> resolutionTables = Maps.newHashMap();
+    for (int resolution : resolutions) {
+      resolutionTables.put(resolution,
+                           tableDef.getDataset(datasetContext, spec.getSpecification(String.valueOf(resolution)),
+                                               arguments, classLoader));
+    }
+
+    Collection<Aggregation> aggregations = getAggregations(spec.getProperties());
+
+    return new CubeDataset(spec.getName(), entityTable, resolutionTables, aggregations);
+  }
+
+  private List<Aggregation> getAggregations(Map<String, String> properties) {
+    // Example of configuring one aggregation with two tags: user and action and user being required:
+    //   dataset.cube.aggregation.1.tags=user,action
+    //   dataset.cube.aggregation.1.requiredTags=user
+
+    Map<String, List<String>> aggTags = Maps.newHashMap();
+    Map<String, Set<String>> aggRequiredTags = Maps.newHashMap();
+    for (Map.Entry<String, String> prop : properties.entrySet()) {
+      if (prop.getKey().startsWith(PROPERTY_AGGREGATION_PREFIX)) {
+        String aggregationProp = prop.getKey().substring(PROPERTY_AGGREGATION_PREFIX.length());
+        String[] nameAndProp = aggregationProp.split("\\.", 2);
+        if (nameAndProp.length != 2) {
+          throw new IllegalArgumentException("Invalid property: " + prop.getKey());
+        }
+        String[] tags = prop.getValue().split(",");
+        if (PROPERTY_TAGS.equals(nameAndProp[1])) {
+          aggTags.put(nameAndProp[0], Arrays.asList(tags));
+        } else if (PROPERTY_REQUIRED_TAGS.equals(nameAndProp[1])) {
+          aggRequiredTags.put(nameAndProp[0], new HashSet<String>(Arrays.asList(tags)));
+        } else {
+          throw new IllegalArgumentException("Invalid property: " + prop.getKey());
+        }
+      }
+    }
+
+    List<Aggregation> aggregations = Lists.newArrayList();
+    for (Map.Entry<String, List<String>> aggTagsEntry : aggTags.entrySet()) {
+      Set<String> requiredTags = aggRequiredTags.get(aggTagsEntry.getKey());
+      requiredTags = requiredTags == null ? Collections.<String>emptySet() : requiredTags;
+      aggregations.add(new DefaultAggregation(aggTagsEntry.getValue(), requiredTags));
```

Comment:
yeah, i was thinking if we have required but not aggregate tags, but we are iterating over aggregate tags, you are right.